### PR TITLE
fix: [CI-6767]: Fix incorrect form validation error for a pipeline with CI stage as template

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

This fixes : https://harness.atlassian.net/browse/CI-6767

https://user-images.githubusercontent.com/48940704/213606680-1ef73235-2e3e-4729-b1c0-1a255dd2a8ce.mov

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by shaurya-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx [hunk 1]
```diff
@@ -396,7 +396,13 @@ function CICodebaseInputSetFormInternal({
               .then((result: ResponseGitBranchesResponseDTO) => {
                 setIsFetchingBranches(false)
                 const branchName = result.data?.defaultBranch?.name || ''
-                formik.setFieldValue(codeBaseInputFieldFormName.branch, branchName)
+                formik?.setValues(
+                  set(
```
Comment (vkbansal-harness) [inline src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:400]: Why 2 `set`s?

Comment (vkbansal-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:400]: better to use `produce` from immerjs

Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:400]: To set each of these fields

image

Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:400]: Currently, only `properties.ci.codebase.build.spec.branch` was getting populated.

Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx]: Done, using `produce` now for mutation.

**Review by vkbansal-harness — APPROVED** [review]

**Review by shaurya-harness — APPROVED** [review]

**Pull request merged**